[WPT import friction] We should use a well-known location for the wpt repo clone that is not inside WebKitBuild

briannafan opened the pull request:
#### 141e123bffe5b5adc3b2899a467f8d2c14b6e696
<pre>
[WPT import friction] We should use a well-known location for the wpt repo clone that is not inside WebKitBuild
<a href="https://bugs.webkit.org/show_bug.cgi?id=277279">https://bugs.webkit.org/show_bug.cgi?id=277279</a>
<a href="https://rdar.apple.com/132753363">rdar://132753363</a>

Reviewed by Sam Sneddon and Tim Nguyen.

This PR changes the default location of the WPT clone to the same dir as the WebKit checkout.
  - For example, if your directory is  ~/WebKit, then we will create ~/wpt.

The import and export scripts will prompt you for a directory to clone WPT into if not specified.
This defaults to the location described above. Notably, the name will default to &apos;wpt&apos; instead of &apos;web-platform-tests&apos;!

* Tools/Scripts/import-webdriver-tests: Use WPTPaths.

* Tools/Scripts/webkitpy/w3c/common.py:
(WPTPaths): Remove &apos;w3c-tests&apos; from the checkout path and use &apos;wpt&apos;.
(WPTPaths.checkout_directory): Use the dir of webkit_base as the checkout directory dir.
(WPTPaths.default_wpt_checkout_path): Rename of wpt_checkout_path.
(WPTPaths.ensure_wpt_repository):
    - This method returns the location of the WPT repo and clones it if it doesn&apos;t exist.
    - If no directory is passed in, it will prompt the user.
    - If run in non-interactive mode, it will default to one level below Webkit.
(WPTPaths.wpt_checkout_path): Deleted.

* Tools/Scripts/webkitpy/w3c/test_exporter.py:
(WebPlatformTestExporter.__init__):
(WebPlatformTestExporter._get_wpt_repository):
(WebPlatformTestExporter.do_export):
(parse_args):
(configure_logging): Use &apos;webkitpy.w3c&apos; logger to pick up common.py logs.
(WebPlatformTestExporter._ensure_wpt_repository):

* Tools/Scripts/webkitpy/w3c/test_exporter_unittest.py:
(TestExporterTest): Remove &apos;WebKitBuild&apos; from the mock path.
(TestExporterTest.test_export):
(TestExporterTest.test_export_with_specific_branch):
(TestExporterTest.test_export_no_clean):
* Tools/Scripts/webkitpy/w3c/test_importer.py:
(TestImporter.do_import):
    - NEW: If a source directory is passed in, we try to clone it if it doesn&apos;t exist.
    - Use WPTPaths.ensure_wpt_repository in interactive mode to clone and return the directory path.

* Tools/Scripts/webkitpy/w3c/test_importer_unittest.py:
    - Add FAKE_WPT_DIR = &apos;/wpt&apos; and replace old dir with this variable.
    - Remove test_checkout_directory since we&apos;re no longer using path_from_webkit_outputdir in common.py.
    - Remove all mentions of &apos;WebKitBuild&apos;.
    - Add mock symlink to setup for each test.
    - Add &apos;wpt&apos; and &apos;testharness.js&apos; files to the mock filesystem to mimic WPT repo.

* Tools/Scripts/webkitpy/w3c/wpt_runner.py:
(WPTRunner.prepare_wpt_checkout): Use WPTPaths.checkout_directory to construct the directory.

* Tools/Scripts/webkitpy/w3c/wpt_runner_unittest.py: Add WPT_CHECKOUT_PATH and replace old paths.
(WPTRunnerTest.test_prepare_wpt_checkout):
(WPTRunnerTest.test_run):
(WPTRunnerTest.test_run_with_args):

Canonical link: <a href="https://commits.webkit.org/300244@main">https://commits.webkit.org/300244@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b64b2b202ad0c5d7ca183c54625323006633f506

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/121663 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/41364 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/32028 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/128181 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/73753 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/0debbd2f-9b30-4ba2-b007-4d819e86cde3) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/123539 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/42076 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/49955 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/92446 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/61471 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/fd80ebde-7850-46d5-a427-0a10427fb928) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/124615 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/33565 "Passed tests") | [⏳ 🧪 api-mac](https://ews-build.webkit.org/#/builders/API-Tests-macOS-EWS "Waiting to run tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/73108 "Found 1 new API test failure: WPE/TestWebKitWebContext:/webkit/WebKitWebContext/no-web-process-leak (failure)") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/e83c5dab-240b-45dd-9997-11472007c881) 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/121020 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/32581 "Passed tests") | [⏳ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/macOS-Sonoma-Release-WK1-Tests-EWS "Waiting to run tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/71698 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/103062 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/27313 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/130978 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/48597 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/36964 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/101016 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/48969 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/105182 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/100907 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/46282 "Passed tests") | [⏳ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/macOS-Sonoma-Release-WK2-Intel-Tests-EWS "Waiting to run tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/45298 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/19292 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/48456 "Built successfully") | | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/47925 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/51274 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/49608 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

nt1m reviewed (commented):
I would drop `w3c-tests` since WPT isn't part of w3c.

Also, I'd change web-platform-tests to wpt since that's the current name of the repo. It's  a bit misleading.

smfr commented on Tools/Scripts/webkitpy/w3c/wpt_runner_unittest.py:
```diff
@@ -26,6 +26,7 @@
 from webkitpy.common.host_mock import MockHost
 from webkitpy.w3c.wpt_runner import WPTRunner, parse_args
 
+WPT_CHECKOUT_PATH = "/w3c-tests/web-platform-tests"
```
Why not just `/wpt`? We should use the same name that cloning https://github.com/web-platform-tests/wpt gives you.

gsnedders replied:
I was gonna argue that https://bugs.webkit.org/show_bug.cgi?id=257730 is a reason not to do that, but then I realised since I opened this to re-review it, I should just write the fix I've been talking about on the shuttle: https://github.com/WebKit/WebKit/pull/49436.

gsnedders commented on Tools/Scripts/webkitpy/common/system/filesystem_mock.py:
```diff
@@ -38,7 +38,7 @@ class MockFileSystem(object):
     pardir = '..'
 
-    def __init__(self, files=None, dirs=None, cwd='/'):
+    def __init__(self, files=None, dirs=None, cwd='/', module_path=None):
```
Nit: I'd make `module_path` default to `"/mock-checkout/Tools/Scripts/"` and thus avoid the branch in `path_to_module`.

gsnedders commented on Tools/Scripts/webkitpy/w3c/common.py:
```diff
@@ -58,13 +58,12 @@ def read_credentials(host, credentials_json):
     @staticmethod
     def wpt_checkout_path(finder):
-        return finder.path_from_webkit_outputdir(*WPTPaths.WPT_CHECKOUT_PATH)
+        return os.path.join(os.path.dirname(finder.webkit_base()), WPTPaths.WPT_CHECKOUT_PATH)
```
Is there any benefit to having this, when it isn't necessarily accurate?

I realise you're still using this to get the default path, but it feels potentially misleading that it doesn't return the "current" path.

My gut would actually be to make this function own the logic you put in `test_exporter.py`; that means changing the signature to something like `wpt_checkout_path(finder: WebKitFinder, *, non_interactive: bool = True) -> str` (and I'd probably just make it raise if non-interactive, at least until we're reading from some config).

We probably also just want to wrap it with `@memotize` to avoid prompting multiple times!

briannafan replied:
Thanks for the feedback! Here are the main changes in the latest update, most of which we've discussed offline already:
- Rename `wpt_checkout_path` to `default_wpt_checkout_path`.
- Create `WPTPaths.ensure_wpt_repository` which now owns the logic and cloning from `test_exporter.py`.
    - This is now also used in `test_importer.py`, and enables cloning into a user given directory.
- If in non-interactive mode, we just clone to the default location. I believe this makes more sense than failing the entire operation.

gsnedders commented on Tools/Scripts/webkitpy/w3c/common.py:
```diff
@@ -58,13 +58,12 @@ def read_credentials(host, credentials_json):
     @staticmethod
     def checkout_directory(finder):
-        return finder.path_from_webkit_outputdir(*WPTPaths.CHECKOUT_DIRECTORY)
+        return os.path.join(os.path.dirname(finder.webkit_base()))
```
Based on my comments re: `wpt_checkout_path`, I think we should just flip which way around they call each other and make this call `wpt_checkout_path` and then find the `dirname` of that.

gsnedders commented on Tools/Scripts/webkitpy/w3c/test_exporter.py:
```diff
@@ -372,7 +380,7 @@ def do_export(self):
     description = f"""Script to generate a pull request to W3C web-platform-tests repository
     'Tools/Scripts/export-w3c-test-changes -c -g HEAD -b XYZ' will do the following:
-    - Clone web-platform-tests repository if not done already and set it up for pushing branches.
+    - Clone web-platform-tests repository if not done already and set it up for pushing branches. The default location is one level below your WebKit checkout.
```
Given we prompt, I don't think we need to add this to the description.

gsnedders reviewed (commented):
> @gsnedders I took your suggestions, could you take a look at `test_exporter.py` especially?

I took a look at everything!

> Also, do we also want this in `test_importer.py`?

Given the bug this PR purports to fix is about the importer, we definitely want the same logic in both places. (Or, per my comments below, shared between them.)

I do still think we want to persist this somewhere, but I don't have a fantastic suggestion of where to persist it. My gut says something like introducing a `.webkitconfig` in the root of the WebKit working tree which we can read and write using the `configparser` module (in the stdlib), and then we can have something like:

```ini
[web-platform-tests]
path = ../web-platform-tests
```

gsnedders commented on Tools/Scripts/import-webdriver-tests:
```diff
@@ -103,7 +104,7 @@ importers = []
 
 if options.w3c:
     importer_config_path = webkit_finder.path_from_webkit_base('WebDriverTests', 'imported', 'w3c', 'importer.json')
-    download_path = webkit_finder.path_from_webkit_base('WebKitBuild', 'w3c-tests', 'web-platform-tests')
+    download_path = WPTPaths.default_wpt_checkout_path(webkit_finder)
```
cc @lauromoura; the changes both here and `wpt_runner.py` are probably of interest to you, given they're how your ports run the WebDriver tests.

gsnedders commented on Tools/Scripts/webkitpy/w3c/common.py:
```diff
@@ -58,13 +60,27 @@ def read_credentials(host, credentials_json):
+
+        if not finder._filesystem.exists(repository_directory):
+            if not os.path.isabs(repository_directory):
+                repository_directory = os.path.abspath(repository_directory)
```
Nit: I wouldn't bother with the `if` here; I'd just always call `abspath` — it's basically a no-op if it's already absolute.

gsnedders commented on Tools/Scripts/webkitpy/w3c/test_exporter.py:
```diff
@@ -29,7 +29,7 @@
 import sys
 
-from webkitcorepy import string_utils, run
+from webkitcorepy import string_utils, run, Terminal
```
`Terminal` is unused now.

gsnedders commented on Tools/Scripts/webkitpy/w3c/common.py:
```diff
@@ -58,13 +60,27 @@ def read_credentials(host, credentials_json):
+                user_directory = Terminal.input(f'Press `Enter` to use the default or input a directory of your choice: ')
+                repository_directory = user_directory.strip() or repository_directory
+
+        if not finder._filesystem.exists(repository_directory):
```
`isdir` and not `exists`; if it exists but it's a file it's not very helpful.

gsnedders commented on Tools/Scripts/webkitpy/w3c/test_importer.py:
```diff
@@ -204,24 +204,20 @@ def __init__(self, port, test_paths, options):
-            if directory is None:
-                return
+        finder = WebKitFinder(self.filesystem)
+        self.source_directory = WPTPaths.ensure_wpt_repository(finder, self.source_directory)
```
This is relying on it being a no-op as long as the directory exists, right?

If I pass in `~/projects`, and I have a `~/projects/web-platform-tests`, I'd probably still expect that to work.

briannafan replied:
Right now if `~/projects` is passed in, we'd literally try `git clone ~/projects`, so your example wouldn't work. Should we be checking for a WPT repository within the source directory passed in? (I want to say no, or at least not in this method.)

The more intuitive assumption is that the source is the location of WPT repo, not the parent, right? That also is in line with how `git clone` behaves.

gsnedders replied:
The code is now `self.source_directory = self.source_directory or WPTPaths.ensure_wpt_repository(finder, self.source_directory)`, so we only run this if not source directory is passed.

This keeps the existing behaviour working, at least.

gsnedders commented on Tools/Scripts/webkitpy/w3c/test_importer.py:
```diff
@@ -204,24 +204,20 @@ def __init__(self, port, test_paths, options):
-        else:
-            _log.info('Downloading W3C test repositories')
-            self.filesystem.maybe_make_directory(self.tests_download_path)
-            self.test_downloader().download_tests(self.options.use_tip_of_tree)
```
This makes a bunch of code in `TestDownloader` dead, I think?

This also means https://bugs.webkit.org/show_bug.cgi?id=231173 is implicitly fixed by this, as it stops defaulting to the out-of-date commit. (I don't know why neither me nor anyone else ever wrote a PR for that…)

briannafan replied:
Indeed it does! Can we make cleaning it up a separate PR?

gsnedders commented on Tools/Scripts/webkitpy/w3c/test_importer_unittest.py:
```diff
@@ -458,19 +460,20 @@ def test_initpy_generation(self):
         host.filesystem = MockFileSystem(files=FAKE_FILES)
 
-        fs = self.import_downloaded_tests(['--no-fetch', '--import-all', '--no-clean-dest-dir', '-d', 'w3c'], FAKE_FILES)
+        with OutputCapture():
```
What are we trying to capture here?

briannafan replied:
Ah, looks like this one has no output. I went through most tests to suppress some git logging/mock process stuff

gsnedders commented on Tools/Scripts/webkitpy/w3c/common.py:
```diff
@@ -58,13 +61,32 @@ def read_credentials(host, credentials_json):
+                return repository_directory
+
+        if run([local.Git.executable(), 'clone', f'{WPT_GH_URL}.git', repository_directory]).returncode:
+            return None
```
Nit:

I wonder how much it makes sense to it has to have this function return `Optional[str]`, rather than having it raising an exception when it fails.

gsnedders commented on Tools/Scripts/webkitpy/w3c/common.py:
```diff
@@ -58,13 +61,32 @@ def read_credentials(host, credentials_json):
+    WPT_CHECKOUT_PATH = "web-platform-tests"
 
     @staticmethod
     def checkout_directory(finder):
```
This PR leaves:

```
webkitpy/w3c/test_importer.py
179:        self.tests_download_path = WPTPaths.checkout_directory(webkit_finder)

webkitpy/w3c/wpt_runner.py
156:            test_downloader = self._downloader_class(WPTPaths.checkout_directory(self._finder),
```

Given the rest of `test_importer.py` has it configurable, I wonder if the former is problematic. I _presume_ not given how long it has supported `--source` for.

The `wpt_runner.py` case feels like we could move — and probably should so we can delete most of `test_downloader.py` — but that is lower priority than the importer/exporter, so probably for follow-up work.

gsnedders replied:
Filed https://bugs.webkit.org/show_bug.cgi?id=299133.

gsnedders commented on Tools/Scripts/webkitpy/w3c/common.py:
```diff
@@ -58,13 +61,32 @@ def read_credentials(host, credentials_json):
     @staticmethod
-    def wpt_checkout_path(finder):
-        return finder.path_from_webkit_outputdir(*WPTPaths.WPT_CHECKOUT_PATH)
+    def default_wpt_checkout_path(finder):
```
This leaves:

```
import-webdriver-tests
107:    download_path = WPTPaths.default_wpt_checkout_path(webkit_finder)

webkitpy/w3c/common.py
68:        return os.path.dirname(WPTPaths.default_wpt_checkout_path(finder))
76:            repository_directory = WPTPaths.default_wpt_checkout_path(finder)

webkitpy/w3c/wpt_runner.py
159:            self._options.wpt_checkout = WPTPaths.default_wpt_checkout_path(self._finder)
```

So the only external usage is the non-Apple WebDriver importer and runner. Which as above, is fine for now.

gsnedders replied:
Filed https://bugs.webkit.org/show_bug.cgi?id=299133.

gsnedders commented on Tools/Scripts/webkitpy/w3c/common.py:
```diff
@@ -58,13 +61,32 @@ def read_credentials(host, credentials_json):
 class WPTPaths:
-    CHECKOUT_DIRECTORY = ["w3c-tests"]
-    WPT_CHECKOUT_PATH = CHECKOUT_DIRECTORY + ["web-platform-tests"]
+    WPT_CHECKOUT_PATH = "web-platform-tests"
```
Per https://github.com/WebKit/WebKit/pull/48812#discussion_r2248420362, can you try changing this to `wpt` and see if everything still passes?

And given this is now only the default, we should rename it like `default_wpt_checkout_path` below (or just get rid of it entirely); it only has two uses — one of them immediately below.

gsnedders commented on Tools/Scripts/webkitpy/w3c/test_importer_unittest.py:
```diff
@@ -103,24 +112,23 @@ def test_import_dir_with_empty_init_py(self):
 
-        host = MockHost()
+        host = MockHost(initialize_scm_by_default=False)
+        host._scm = MockSCM(filesystem=host.filesystem, executive=host.executive, checkout_root='/home/mock-checkout')
```
This appears to be unneeded?

gsnedders commented on Tools/Scripts/webkitpy/w3c/test_importer_unittest.py:
```diff
@@ -130,15 +138,16 @@ def test_import_dir_not_web_platform_tests(self):
     def import_downloaded_tests(self, args, files, test_port=False):
-        host = MockHost()
+        host = MockHost(initialize_scm_by_default=False)
+        host._scm = MockSCM(filesystem=host.filesystem, executive=host.executive, checkout_root='/home/mock-checkout')
```
This appears to be unneeded?

gsnedders commented on Tools/Scripts/webkitpy/w3c/test_importer_unittest.py:
```diff
@@ -33,6 +33,7 @@
 from webkitcorepy import OutputCapture
 from webkitscmpy.mocks.local import Git as Git_mock
 
+from webkitpy.common.checkout.scm.scm_mock import MockSCM
```
This appears to be unneeded?

gsnedders commented on Tools/Scripts/webkitpy/w3c/test_importer_unittest.py:
```diff
@@ -67,6 +71,11 @@
 
 
 class TestImporterTest(unittest.TestCase):
+    def setUp(self):
```
Can you preserve my old comment: "This is a horrible hack so we don't actually need to support symlinks in the MockFileSystem."

gsnedders commented on Tools/Scripts/webkitpy/w3c/common.py:
```diff
@@ -58,13 +61,32 @@ def read_credentials(host, credentials_json):
+    def ensure_wpt_repository(finder, repository_directory=None, *, non_interactive=True):
+        if not repository_directory:
+            repository_directory = WPTPaths.default_wpt_checkout_path(finder)
+            print(f'The WPT repository will be cloned by default into `{repository_directory}`.')
```
Why `print` and not `_log.info`?

briannafan replied:
Probably just habit. Changing to `_log.info` to match the rest of the file.

briannafan replied:
Actually, after testing, `_log.info` won't expose the log when running this from `export-w3c-test-changes`..

gsnedders replied:
This seems better fixed by changing `logger = logging.getLogger('webkitpy.w3c.test_exporter')` to `webkitpy.w3c` in `test_exporter.py`?

With that, I then think we don't need the `OutputCapture` in the tests?

gsnedders commented on Tools/Scripts/webkitpy/w3c/common.py:
```diff
@@ -58,13 +61,32 @@ def read_credentials(host, credentials_json):
+
+        if finder._filesystem.isdir(repository_directory):
+            if finder._filesystem.isfile(str(d / "resources" / "testharness.js")) and finder._filesystem.isfile(str(d / "wpt")):
+                print(f'Using the WPT repository found at {repository_directory}')
```
Why `print` and not `_log.info`?

gsnedders commented on Tools/Scripts/webkitpy/w3c/test_importer_unittest.py:
```diff
@@ -41,19 +42,22 @@
 FAKE_FILES = {
     '/tests/csswg/css-fake-1/empty_dir/README.txt': '',
-    '/mock-checkout/LayoutTests/w3c/css-fake-1/README.txt': '',
+    '/home/mock-checkout/LayoutTests/w3c/css-fake-1/README.txt': '',
```
The commit message says "Initialize MockSCM with '/home/mock-checkout' so WPT will clone to a non-root directory." — but nothing uses the old webkitpy SCM code as far as I'm aware.

Trying this locally, changing `/home/mock-checkout` back to `/mock-checkout` and changing `FAKE_WPT_DIR` to `/web-platform-tests` (i.e., the default found path, adjacent to `/mock-checkout`) works fine?

gsnedders commented on Tools/Scripts/webkitpy/w3c/test_importer_unittest.py:
```diff
@@ -41,19 +42,22 @@
-    '/mock-checkout/LayoutTests/imported/w3c/resources/resource-files.json': '{"directories": [], "files": []}',
+    '/home/mock-checkout/LayoutTests/imported/w3c/resources/resource-files.json': '{"directories": [], "files": []}',
+    '/home/web-platform-tests/wpt': '',
+    '/home/web-platform-tests/resources/testharness.js': '',
```
```suggestion
    f'{FAKE_WPT_DIR}/wpt': '',
    f'{FAKE_WPT_DIR}/resources/testharness.js': '',
```

gsnedders commented on Tools/Scripts/webkitpy/w3c/wpt_runner.py:
```diff
@@ -156,7 +156,7 @@ def prepare_wpt_checkout(self):
                 self._host, self._downloader_class.default_options())
             test_downloader.clone_tests()
-            self._options.wpt_checkout = WPTPaths.wpt_checkout_path(self._finder)
+            self._options.wpt_checkout = WPTPaths.default_wpt_checkout_path(self._finder)
```
Because this is still using `test_downloader.clone_tests()`, this actually needs to be `fs.join(WPTPaths.checkout_directory(self._finder), "web-platform-tests")`.

gsnedders reviewed (changes requested):
A decent number of these are "we don't need this change any more with what this PR now is".

And because I was trying to see what broke if you _didn't_ change them, I can offer you https://github.com/briannafan/WebKit/pull/1 which addresses many but not all of the below comments.

gsnedders commented on Tools/Scripts/webkitpy/w3c/test_importer_unittest.py:
```diff
@@ -103,24 +111,22 @@ def test_import_dir_with_empty_init_py(self):
 
-        host = MockHost()
+        host = MockHost(initialize_scm_by_default=False)
+        host.filesystem.module_path = "/mock-checkout/Tools/Scripts/"
```
This isn't used any more.

gsnedders commented on Tools/Scripts/webkitpy/w3c/test_importer_unittest.py:
```diff
@@ -130,15 +136,15 @@ def test_import_dir_not_web_platform_tests(self):
     def import_downloaded_tests(self, args, files, test_port=False):
-        host = MockHost()
+        host = MockHost(initialize_scm_by_default=False)
+        host.filesystem.module_path = "/mock-checkout/Tools/Scripts/"
```
This isn't used any more.

gsnedders commented on Tools/Scripts/webkitpy/w3c/test_importer_unittest.py:
```diff
@@ -149,21 +155,23 @@ def import_downloaded_tests(self, args, files, test_port=False):
 
         options, test_paths = parse_args(args)
         importer = TestImporter(port, test_paths, options)
+        fs.maybe_make_directory(fs.join(importer.tests_download_path, "web-platform-tests"))
```
Is this necessary?

briannafan replied:
Not anymore!

briannafan committed:
[WPT import friction] We should use a well-known location for the wpt repo clone that is not inside WebKitBuild
https://bugs.webkit.org/show_bug.cgi?id=277279
rdar://132753363

Reviewed by Sam Sneddon and Tim Nguyen.

This PR changes the default location of the WPT clone to the same dir as the WebKit checkout.
  - For example, if your directory is  ~/WebKit, then we will create ~/wpt.

The import and export scripts will prompt you for a directory to clone WPT into if not specified.
This defaults to the location described above. Notably, the name will default to 'wpt' instead of 'web-platform-tests'!

* Tools/Scripts/import-webdriver-tests: Use WPTPaths.

* Tools/Scripts/webkitpy/w3c/common.py:
(WPTPaths): Remove 'w3c-tests' from the checkout path and use 'wpt'.
(WPTPaths.checkout_directory): Use the dir of webkit_base as the checkout directory dir.
(WPTPaths.default_wpt_checkout_path): Rename of wpt_checkout_path.
(WPTPaths.ensure_wpt_repository):
    - This method returns the location of the WPT repo and clones it if it doesn't exist.
    - If no directory is passed in, it will prompt the user.
    - If run in non-interactive mode, it will default to one level below Webkit.
(WPTPaths.wpt_checkout_path): Deleted.

* Tools/Scripts/webkitpy/w3c/test_exporter.py:
(WebPlatformTestExporter.__init__):
(WebPlatformTestExporter._get_wpt_repository):
(WebPlatformTestExporter.do_export):
(parse_args):
(configure_logging): Use 'webkitpy.w3c' logger to pick up common.py logs.
(WebPlatformTestExporter._ensure_wpt_repository):

* Tools/Scripts/webkitpy/w3c/test_exporter_unittest.py:
(TestExporterTest): Remove 'WebKitBuild' from the mock path.
(TestExporterTest.test_export):
(TestExporterTest.test_export_with_specific_branch):
(TestExporterTest.test_export_no_clean):
* Tools/Scripts/webkitpy/w3c/test_importer.py:
(TestImporter.do_import):
    - NEW: If a source directory is passed in, we try to clone it if it doesn't exist.
    - Use WPTPaths.ensure_wpt_repository in interactive mode to clone and return the directory path.

* Tools/Scripts/webkitpy/w3c/test_importer_unittest.py:
    - Add FAKE_WPT_DIR = '/wpt' and replace old dir with this variable.
    - Remove test_checkout_directory since we're no longer using path_from_webkit_outputdir in common.py.
    - Remove all mentions of 'WebKitBuild'.
    - Add mock symlink to setup for each test.
    - Add 'wpt' and 'testharness.js' files to the mock filesystem to mimic WPT repo.

* Tools/Scripts/webkitpy/w3c/wpt_runner.py:
(WPTRunner.prepare_wpt_checkout): Use WPTPaths.checkout_directory to construct the directory.

* Tools/Scripts/webkitpy/w3c/wpt_runner_unittest.py: Add WPT_CHECKOUT_PATH and replace old paths.
(WPTRunnerTest.test_prepare_wpt_checkout):
(WPTRunnerTest.test_run):
(WPTRunnerTest.test_run_with_args):

Canonical link: https://commits.webkit.org/300244@main
